src/providers/spatialite/qgsspatialitesourceselect.cpp

```diff
@@ -312,6 +314,11 @@ bool QgsSpatiaLiteSourceSelect::newConnection( QWidget *parent )
   // inserting this SQLite DB path
   settings.setValue( baseKey + "selected", savedName );
   settings.setValue( baseKey + savedName + "/sqlitepath", myFI.canonicalFilePath() );
+
+  QgsProviderMetadata *providerMetadata = QgsProviderRegistry::instance()->providerMetadata( QStringLiteral( "spatialite" ) );
+  QgsSpatiaLiteProviderConnection *providerConnection =  static_cast<QgsSpatiaLiteProviderConnection *>( providerMetadata->createConnection( savedName ) );
+  providerMetadata->saveConnection( providerConnection, savedName );
+
```

Comment:
won't the call to QgsSpatiaLiteConnection::deleteConnection in btnDelete_clicked also need to be fixed? (and then QgsSpatiaLiteConnection::deleteConnection can be totally removed -- it's not part of stable api, and will be unused code)

Reply:
Indeed, I removed it after the changes on btnDelete_clicked